pkg/service/restore/worker.go

```diff
@@ -224,6 +231,45 @@ func IsRestoreSchemaSupported(ctx context.Context, client *scyllaclient.Client)
 	return nil
 }
 
+// ErrRestoreAuthAndServiceLevelsUnsupportedScyllaVersion means that restore auth and service levels procedure is not safe for used Scylla configuration.
+var ErrRestoreAuthAndServiceLevelsUnsupportedScyllaVersion = errors.Errorf("restoring authentication and service levels is not supported for given ScyllaDB version")
+
+// IsRestoreAuthAndServiceLevelsFromSStablesSupported checks if restore auth and service levels procedure is supported for used Scylla configuration.
+// Because of #3869 and #3875, there is no way fo SM to safely restore auth and service levels into cluster with
+// version higher or equal to OSS 6.0 or ENT 2024.2.
+func IsRestoreAuthAndServiceLevelsFromSStablesSupported(ctx context.Context, client *scyllaclient.Client) error {
+	const (
+		ossConstraint = ">= 6.0, < 2000"
+		entConstraint = ">= 2024.2, > 1000"
+	)
+
+	status, err := client.Status(ctx)
+	if err != nil {
+		return errors.Wrap(err, "get status")
+	}
+	for _, n := range status {
+		ni, err := client.NodeInfo(ctx, n.Addr)
```

Comment:
NodeInfo is a concatenation of different API calls to Scylla. Can be simplified with direct call for scylla_version.

Reply:
That's true. A similar situation happens in the `IsRestoreSchemaFromSSTablesSupported` method, but there we also need the `consistent_cluster_management` config field.

What would you say if we made `configcache` svc reachable from `restore` svc (the same as in `healthcheck` svc) and use `nodeInfo` already fetched by `configcache` svc instead of doing it on our own? This would also help in other places that fetch `nodeInfo`.

I could do it as a part of a refactoring PR, but I would leave this as it is in the current one.

Reply:
I would prefer to not spread the configcache among all the services yet.
We must be sure that everything is fine with it and it's battle-tested for some time.

But, it makes sense to leave it as is for the moment and refactor in future.